improver_tests/utilities/temporal/test_TimezoneExtraction.py

```diff
@@ -0,0 +1,259 @@
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for TimezoneExtraction plugin."""
+
+from datetime import datetime, timedelta
+
+import iris
+import numpy as np
+import pytest
+from cf_units import Unit
+from iris.cube import Cube
+
+from improver.metadata.check_datatypes import check_mandatory_standards
+from improver.metadata.constants.time_types import TIME_COORDS
+from improver.synthetic_data.set_up_test_cubes import (
+    add_coordinate,
+    set_up_variable_cube,
+)
+from improver.utilities.temporal import TimezoneExtraction
+
+
+def make_input_cube(data_shape, time_bounds=True):
+    """Makes a 3D cube (time, y, x) of the described shape, filled with zeroes for use
+    in testing."""
+    cube = set_up_variable_cube(
+        np.zeros(data_shape).astype(np.float32),
+        standard_grid_metadata="gl_ens",
+        attributes={
+            "institution": "unknown",
+            "source": "IMPROVER",
+            "title": "Unit test",
+        },
+    )
+    cube = add_coordinate(
+        cube,
+        [datetime(2017, 11, 10, 4, 0) + timedelta(hours=h) for h in range(3)],
+        "time",
+        coord_units=TIME_COORDS["time"].units,
+        dtype=TIME_COORDS["time"].dtype,
+        is_datetime=True,
+    )
+    if time_bounds:
+        cube.coord("time").bounds = np.array(
+            [
+                [
+                    np.around(
+                        Unit(TIME_COORDS["time"].units).date2num(
+                            datetime(2017, 11, 10, 4, 0) + timedelta(hours=h + b)
+                        )
+                    )
+                    for b in [-1, 0]
+                ]
+                for h in range(3)
+            ],
+            dtype=TIME_COORDS["time"].dtype,
+        )
+    return cube
+
+
+def make_timezone_cube():
+    """Makes a timezone cube to use in tests. data=0 where points fall in that
+    time-zone. data=1 where they don't."""
+    cube = set_up_variable_cube(
+        np.zeros((3, 3)).astype(np.float32),
+        name="timezone_mask",
+        units="1",
+        standard_grid_metadata="gl_ens",
+        attributes={
+            "institution": "unknown",
+            "source": "IMPROVER",
+            "title": "Unit test",
+        },
+    )
+    cube = add_coordinate(
+        cube,
+        [-3600, 0, 3600],
+        "UTC_offset",
+        coord_units="seconds",
+        dtype=TIME_COORDS["forecast_period"].dtype,
+    )
+    true_row = [1, 1, 1]
+    false_row = [0, 0, 0]
+    cube.data = np.array(
+        [
+            [true_row, true_row, false_row],
+            [true_row, false_row, true_row],
+            [false_row, true_row, true_row],
+        ],
+        dtype=np.int8,
+    )
```

Comment:
This is a weird timezone cube in which the timezones change with latitude rather than longitude. I don't think it matters for the tests, but it looks a bit funny.